From e9d265cb53a3cac8a31c612462694be09823333f Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Mon, 9 Jan 2012 11:07:12 +0530
Subject: [PATCH 541/609] pwm: Platform data is placed in common position

Original commit: 811c1b31b8d951864b9bd02b1636b216a6dd53ec

  From git://arago-project.org/git/projects/linux-am33x.git
  And ti-sdk-am335x-evm-05.05.00.00-Linux-x86-Install image

Platform_driver_data  for eHRPWM module is put to common place to access
the structure irrespective of architecture.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 drivers/pwm/ehrpwm.c       |    3 ++-
 include/linux/pwm/ehrpwm.h |    4 ----
 include/linux/pwm/pwm.h    |    4 ++++
 3 files changed, 6 insertions(+), 5 deletions(-)

diff --git a/drivers/pwm/ehrpwm.c b/drivers/pwm/ehrpwm.c
index 91893dd..879abf5 100644
--- a/drivers/pwm/ehrpwm.c
+++ b/drivers/pwm/ehrpwm.c
@@ -20,6 +20,7 @@
 #include <linux/interrupt.h>
 #include <linux/platform_device.h>
 #include <linux/slab.h>
+#include <linux/pwm/pwm.h>
 #include <linux/pwm/ehrpwm.h>
 
 #include <plat/clock.h>
@@ -1291,7 +1292,7 @@ static int ehrpwm_probe(struct platform_device *pdev)
 	struct resource *r;
 	int ret = 0;
 	int chan = 0;
-	struct ehrpwm_platform_data *pdata = (&pdev->dev)->platform_data;
+	struct pwmss_platform_data *pdata = (&pdev->dev)->platform_data;
 	int ch_mask;
 
 	ch_mask = pdata->channel_mask;
diff --git a/include/linux/pwm/ehrpwm.h b/include/linux/pwm/ehrpwm.h
index eb87516..d41e6f2 100644
--- a/include/linux/pwm/ehrpwm.h
+++ b/include/linux/pwm/ehrpwm.h
@@ -5,10 +5,6 @@
 
 #define NCHAN 2
 
-struct ehrpwm_platform_data {
-	int channel_mask;
-};
-
 struct ehrpwm_pwm;
 
 typedef int (*p_fcallback) (struct ehrpwm_pwm *, void *data);
diff --git a/include/linux/pwm/pwm.h b/include/linux/pwm/pwm.h
index f912c25..cebea99 100644
--- a/include/linux/pwm/pwm.h
+++ b/include/linux/pwm/pwm.h
@@ -105,6 +105,10 @@ struct pwm_device {
 	spinlock_t pwm_lock;
 };
 
+struct pwmss_platform_data {
+	int channel_mask;
+};
+
 struct pwm_device *pwm_request_byname(const char *name, const char *label);
 struct pwm_device *pwm_request(const char *bus_id, int id, const char *label);
 void pwm_release(struct pwm_device *p);
-- 
1.7.5.4

